lib/view_component/compiler.rb

```diff
@@ -192,14 +192,23 @@ def templates
       @templates ||=
         begin
           extensions = ActionView::Template.template_handler_extensions
-
-          component_class._sidecar_files(extensions).each_with_object([]) do |path, memo|
-            pieces = File.basename(path).split(".")
-            memo << {
-              path: path,
-              variant: pieces.second.split("+").second&.to_sym,
-              handler: pieces.last
-            }
+          check_class = component_class
+
+          loop do
```

Comment:
Is it possible to avoid a `loop` here? I generally tend to avoid non-bounded loops since it makes it easier to introduce performance regressions or infinite loop errors.

Maybe we could look at the classes `ancestors` array instead?